File: 2018/1/1.2.py
import sys

freqs = {}

freq = 0
with open("./input") as input:
    input = list(input)
    while True:
        for l in input:
            freq += int(l)

            if freq in freqs:
                print(freq)
                sys.exit(0)
            else:
                freqs[freq] = 1

File: 2018/1/1.2.clj
(use 'clojure.java.io)

(with-open [r (reader "./input")]

  (let [deltas (->> r
                    line-seq
                    (map #(Integer/parseInt %))
                    cycle)]

    (defn find-repeat [deltas]
      (loop [freq 0
             deltas deltas
             seen #{}]
        (let [nfreq (+ freq (first deltas))]
          (if (contains? seen nfreq)
            nfreq
            (recur nfreq (rest deltas) (conj seen nfreq))))))

    (print (find-repeat deltas))))
